### PR TITLE
Feature 4445 surv net interface care accommodation work in facility android

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 1]
```diff
@@ -1969,17 +1973,28 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
...
+					"UPDATE cases SET screeningType = 'EDUCATIONAL_INSTITUTIONS', caseIdentificationSource = 'SCREENING' where caseIdentificationSource = 'EDUCATIONAL_INSTITUTIONS';");
 
 			case 277:
 				currentVersion = 277;
 				getDao(Location.class).executeRaw("ALTER TABLE location ADD COLUMN country_id BIGINT REFERENCES country(id);");
 
+			case 278:
+				currentVersion = 278;
+				TableUtils.createTable(connectionSource, ActivityAsCase.class);
+				getDao(EpiData.class).executeRaw("ALTER TABLE epidata ADD COLUMN activityascasedetailsknown varchar(255);");
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:1996]: This upgrade script does not work. SQLite is case sensitive and doesn't recognise this field as activityAsCaseDetailsKnown.

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/activityascase/ActivityAsCaseDao.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	@Override
+	public void deleteCascade(ActivityAsCase activityAsCase) throws SQLException {
+		DatabaseHelper.getLocationDao().delete(activityAsCase.getLocation());
+		super.delete(activityAsCase);
+	}
+
+	public List<ActivityAsCase> getByEpiData(EpiData epiData) {
+		if (epiData.isSnapshot()) {
+			return querySnapshotsForEq(Exposure.EPI_DATA + "_id", epiData, ActivityAsCase.CHANGE_DATE, false);
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/activityascase/ActivityAsCaseDao.java:51]: Done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 3]
```diff
@@ -15,19 +15,6 @@
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:15]: Done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/epidata/EpiDataDao.java [hunk 4]
```diff
@@ -90,7 +93,9 @@ public Date getLatestChangeDate() {
 		}
 
 		Date exposureDate = DatabaseHelper.getExposureDao().getLatestChangeDate();
-		if (exposureDate != null && exposureDate.after(date)) {
+		Date activityAsCaseDate = DatabaseHelper.getActivityAsCaseDao().getLatestChangeDate();
+
+		if (exposureDate != null && (exposureDate.after(date) || activityAsCaseDate.after(date))) {
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/epidata/EpiDataDao.java:98]: I agreed, I used your suggestion

File: sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/ActivityAsCaseDialog.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		contentBinding.activityAsCaseRole.initializeSpinner(DataUtils.getEnumItems(ExposureRole.class, true));
+		contentBinding.activityAsCaseWorkEnvironment.initializeSpinner(DataUtils.getEnumItems(WorkEnvironment.class, true));
+
+		contentBinding.activityAsCaseActivityAsCaseType.addValueChangedListener(e -> {
+		});
+		contentBinding.activityAsCaseMeansOfTransport.addValueChangedListener(e -> {
+			contentBinding.activityAsCaseConnectionNumber.setCaption(
+				e.getValue() == MeansOfTransport.PLANE
+					? I18nProperties.getCaption(Captions.exposureFlightNumber)
+					: I18nProperties.getPrefixCaption(ExposureDto.I18N_PREFIX, ExposureDto.CONNECTION_NUMBER));
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/ActivityAsCaseDialog.java:112]: changed

File: sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/ActivityAsCaseDialog.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+		});
+
+		contentBinding.activityAsCaseLocation.setOnClickListener(v -> openAddressPopup());
+
+		setFieldVisibilitiesAndAccesses(ExposureDto.class, (ViewGroup) getRootView());
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/ActivityAsCaseDialog.java:117]: changed, also found another issue like this above

File: sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/EpidemiologicalDataReadFragment.java [hunk 7]
```diff
@@ -104,6 +108,36 @@ private void setUpControlListeners() {
...
+				item,
+				boundView -> FieldVisibilityAndAccessHelper.setFieldVisibilitiesAndAccesses(
+					ActivityAsCaseDto.class,
+					(ViewGroup) boundView,
+					new FieldVisibilityCheckers(),
+					getFieldAccessCheckers()));
+
+			final DialogActivityAsCaseReadLayoutBinding activityAsCaseBinding = (DialogActivityAsCaseReadLayoutBinding) infoDialog.getBinding();
+			if (((ActivityAsCase) item).getMeansOfTransport() == MeansOfTransport.PLANE) {
+				activityAsCaseBinding.activityAsCaseConnectionNumber.setCaption(I18nProperties.getCaption(Captions.exposureFlightNumber));
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/epidata/EpidemiologicalDataReadFragment.java:125]: changed

File: sormas-app/app/src/main/res/layout/dialog_activity_as_case_edit_layout.xml [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+                style="@style/ControlSingleColumnStyle" />
+
+            <de.symeda.sormas.app.component.controls.ControlTextEditField
+                android:id="@+id/activityAsCase_habitationDetails"
+                app:value="@={data.habitationDetails}"
+                app:dependencyParentField="@{activityAsCaseHabitationType}"
+                app:dependencyParentValue="@{HabitationType.OTHER}"
+                style="@style/ControlSingleColumnStyle" />
+
+            <TextView
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/res/layout/dialog_activity_as_case_edit_layout.xml:122]: removed

File: sormas-app/app/src/main/res/layout/dialog_activity_as_case_read_layout.xml [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+                    <de.symeda.sormas.app.component.controls.ControlTextReadField
+                        android:id="@+id/activityAsCase_habitationDetails"
+                        app:value="@{data.habitationDetails}"
+                        app:dependencyParentField="@{activityAsCaseHabitationType}"
+                        app:dependencyParentValue="@{HabitationType.OTHER}"
+                        style="@style/ControlSecondOfTwoColumnsStyle" />
+
+                </LinearLayout>
+
+                <TextView
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/res/layout/dialog_activity_as_case_read_layout.xml:162]: removed

File: sormas-app/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -377,6 +377,7 @@
     <string name="heading_extend_followup">Extend follow-up period</string>
     <string name="heading_contact_tracing_first_contact">First contact with Contact Tracing</string>
     <string name="heading_exposure">Exposure</string>
+    <string name="heading_activityAsCase">Activity</string>
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/res/values/strings.xml:380]: updated

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:1996]: updated the field name

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/epidata/EpiDataDao.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/epidata/EpiDataDao.java]: I've made it shorter

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**